zebra-state/src/service/finalized_state/disk_format/upgrade/add_subtrees.rs

```diff
@@ -353,7 +353,12 @@ fn check_sapling_subtrees(db: &ZebraDb) -> Result<(), &'static str> {
         }
         // Check that the final note has a greater subtree index if it didn't complete a subtree.
         else {
-            let Some(prev_tree) = db.sapling_tree_by_height(&subtree.end.previous()) else {
+            let prev_height = subtree
+                .end
+                .previous()
+                .expect("Height should not underflow here.");
```

Comment:
Usually we put a reason in the `expect()` string, so it's easier to fix bugs.

```suggestion
                .expect("underflow is impossible: Sapling activation is a long way from genesis");
```

Reply:
This applies to all the `expect()` and `unwrap()` in this PR.